status/condition_set.go

```diff
@@ -237,3 +239,60 @@ func (c ConditionSet) findUnhealthyDependents() []Condition {
 	})
 	return conditions
 }
+
+// objectImpl is an embedded struct type that embeds the upstream type
+type objectImpl struct {
+	unstructured.Unstructured
```

Comment:
```suggestion
	*unstructured.Unstructured
```

Should we store this as a pointer since we are receiving a pointer from the `FromUnstructured` call?